src/EFTools/EntityDesignEntityDesigner/CustomCode/Shapes/EntityTypeShape.cs

```diff
@@ -690,28 +687,35 @@ public override string AccessibleHelp
             }
         }
 
-        public override void OnKeyUp(DiagramKeyEventArgs e)
+        private void EntityTypeShape_KeyUp(object sender, DiagramKeyEventArgs e)
         {
             if (e.Control && e.KeyCode == Keys.Up)
             {
-                var isExpanded = _expandCollapseField.IsExpanded(this);
-                if (isExpanded.HasValue && isExpanded.Value)
+                if (IsExpanded)
```

Comment:
Nit: I feel tempted to try to refactor this to have a single extension method, e.g. something like EnsureExpandedState(this NodeShape shape, bool state) that could be reused for compartments and entities. Not sure it is possible, and it is not really necessary, just a matter of reducing code duplication.

Reply:
Agreed. I'll try to find a good place to put it and call it. But if it ends up taking too long I'll just revert to this. I don't want to spend too much time on it, this way is clear to read and it's unlikely we will have to revisit or generalize the collapse/expand code any time in the future.